packages/react-core/src/components/Alert/examples/Alert.md

```diff
@@ -13,9 +13,16 @@ import DatabaseIcon from '@patternfly/react-icons/dist/esm/icons/database-icon';
 import ServerIcon from '@patternfly/react-icons/dist/esm/icons/server-icon';
 import LaptopIcon from '@patternfly/react-icons/dist/esm/icons/laptop-icon';
 
-## Examples
+## Component overview
+
+[`Alert`](/components/alert#alert) can be paired with the following components to allow for additional functionality and customization: [`AlertActionCloseButton`](/components/alert#alertactionclosebutton) and [`AlertActionLink`](/components/alert#alertactionlink).
```

Comment:
I wasn't sure if we wanted to do that, or if adding the context down by the components/properties was the goal, as seen here https://patternfly-react-pr-7603.surge.sh/components/file-upload---multiple/#multiplefileuploadstatusitem. Couldn't remember what the plans were for that example, but I could go ahead and add explanation up top if that's the better route for now

Reply:
With how many sub-components that could possibly get listed on some other components (Table being one that comes to mind immediately), I think explaining what the additional components are for at the top of the page might end up being too verbose. It might depend what the "Component overview" section should accomplish, though. If we'd want to list out sub-components that can be used alongside the Alert and what they are for, it would be worth aligning on how we can go about that without duplicating information.

A couple of potential ways to go about it that I could think of:

- "Component overview" includes only the names of sub-components and a `"For more information about these sub-components, read the [alert props](...) documentation"` (rather than having links to each individual component, mainly thinking about whether it would be obvious what exactly the first "Alert" link is linking to for users navigating via screen reader)
- Include a more general description in this "Component overview" section, and go into more detail in the "Props" section

Reply:
I'm in the same boat about not being sure what answer is right/better. I think it may ultimately depend how the "story" should flow for the React (and later, HTML/CSS) tab and what purpose the "component overview" section should have. Taking into account how a component page may end up looking after some planned changes, below is one possibility I thought of.

Looking at the HTML tab for components, there is a "documentation" section which houses an "overview" and "usage" section, which the "usage" section seems similar to the React "props" section. Renaming that "props" section was brought up so one possibility is renaming that section to "usage" on the React tab. This "usage" section would essentially be this "component overview" section, then from there it would be a matter of should examples or "usage" come first on a component page. The structure of a component's React tab _could_ look like:

- Component name
- Description
- Table of Contents
- Examples (or Overview)
    - All of the examples
- Overview (or Examples)
    - Possibly the "Composable structure" section when required, as seen in the PR Erin linked above, though it might depend if it makes more sense to keep this closer to any composable examples
    - Usage; would essentially be the current props section, but with the description of the different components (if there are multiple)
- CSS Variables

@kmcfaul @edonehoo what do you both think?

Reply:
I'm a bit divided too since I'm not sure what users utilize the most on these pages. Do most people use this page to see specific examples or to quickly reference subcomponents and props? The prop tables seem to mostly just be referential information, so my inclination is that people are more interested in examples, but I'm curious if you all know what people are actually doing? 

As long as referencing/linking components and props won't be too confusing for the average user, I think it may make sense for examples to come first: 
- Component name
- Description
- Table of Contents
- Examples 
- All of the examples
- Overview 
- Possibly the "Composable structure" section when required
- Usage
- CSS Variables

I feel like the examples can be understood without first reading a props overview, especially if people just want to grab the code snippet and leave. This would also keep the component/prop context closer to the tables, so people won't have to scroll back and forth?

Reply:
@edonehoo I'd agree with what you said. I think ultimately there may be a pro/con to either way we go; either examples come first and some users are missing context of what X component or Y prop does until they continue reading further down, or users see explanations of components/props first without the context of how they all come together in examples/when they may just want the examples.

The way I might view it is the props table, component descriptions, and to an extent composable structure would be information that users can refer to when they want to use a component beyond what any of our examples show (or just for users who are interested in learning more). Whether that *is* the case, i.e. users are utilizing this info more than just grabbing the example code, I'm not sure.